netdb.c

```diff
@@ -0,0 +1,264 @@
+		} else if (type == AF_INET6) {
+			struct in6_addr *in6_addr = (struct in6_addr *) addr;
+
+			inet_ntop(AF_INET, in6_addr, ip_addr, sizeof(ip_addr));
+		}
+
+		if (hent)
+			trace_printf(1, "gethostbyaddr(IP => %s, , type => %s) = [Hostname:%s]\n", ip_addr,
+				type == AF_INET ? "AF_INET" : "AF_INET6",
+				hent->h_name);
+		else
+			trace_printf(1, "gethostbyaddr(IP => %s, , type => %s) = [Error:%s]\n", ip_addr,
+				type == AF_INET ? "AF_INET" : "AF_INET6",
+				hstrerror(h_errno));
+	}
+
+	return hent;
+}
+
+RETRACE_REPLACE(gethostbyaddr);
+
+void RETRACE_IMPLEMENTATION(sethostent)(int stayopen)
+{
+	rtr_sethostent_t real_sethostent;
+
+	real_sethostent = RETRACE_GET_REAL(sethostent);
+
+	trace_printf(1, "sethostent(%d)\n", stayopen);
+
+	return real_sethostent(stayopen);
+}
+
+RETRACE_REPLACE(sethostent);
+
+void RETRACE_IMPLEMENTATION(endhostent)(void)
+{
+	rtr_endhostent_t real_endhostent;
+
+	real_endhostent = RETRACE_GET_REAL(endhostent);
+
+	trace_printf(1, "endhostent()\n");
+
+	return real_endhostent();
+}
+
+RETRACE_REPLACE(endhostent);
+
+struct hostent *RETRACE_IMPLEMENTATION(gethostent)(void)
+{
+	rtr_gethostent_t real_gethostent;
+	struct hostent *hent;
+
+	real_gethostent = RETRACE_GET_REAL(gethostent);
+
+	hent = real_gethostent();
+	if (hent) {
+		struct in_addr **addr_list;
+		int i;
+
+		trace_printf(1, "gethostent() = [");
+
+		addr_list = (struct in_addr **) hent->h_addr_list;
+		for (i = 0; addr_list[i] != NULL; i++) {
+			char ip_addr[INET6_ADDRSTRLEN];
+
+			inet_ntop(hent->h_addrtype, addr_list[i], ip_addr, sizeof(ip_addr));
+			if (i > 0)
+				trace_printf(0, ",");
+			trace_printf(0, "%s", ip_addr);
+		}
+
+		trace_printf(0, "]\n");
+	} else
+		trace_printf(1, "gethostent() = NULL [error:%s]\n", hstrerror(h_errno));
+
+	return hent;
+}
+
+RETRACE_REPLACE(gethostent);
+
+struct hostent *RETRACE_IMPLEMENTATION(gethostbyname2)(const char *name, int af)
+{
+	rtr_gethostbyname2_t real_gethostbyname2;
+	struct hostent *hent;
+
+	real_gethostbyname2 = RETRACE_GET_REAL(gethostbyname2);
+
+	hent = real_gethostbyname2(name, af);
+	if (hent) {
+		trace_printf(1, "gethostbyname2(\"%s\", %s) = [", name, af == AF_INET ? "AF_INET" : "AF_INET6");
+
+		if (af == AF_INET) {
+			struct in_addr **addr_list;
+			int i;
+
+			addr_list = (struct in_addr **) hent->h_addr_list;
+			for (i = 0; addr_list[i] != NULL; i++) {
+				char ip_addr[INET6_ADDRSTRLEN];
+
+				inet_ntop(hent->h_addrtype, addr_list[i], ip_addr, sizeof(ip_addr));
+				if (i > 0)
+					trace_printf(0, ",");
+				trace_printf(0, "%s", ip_addr);
+			}
+		} else if (af == AF_INET6) {
+			struct in6_addr **addr_list;
+			int i;
+
+			addr_list = (struct in6_addr **) hent->h_addr_list;
+			for (i = 0; addr_list[i] != NULL; i++) {
+				char ip_addr[INET6_ADDRSTRLEN];
+
+				inet_ntop(hent->h_addrtype, addr_list[i], ip_addr, sizeof(ip_addr));
+				if (i > 0)
+					trace_printf(0, ",");
+				trace_printf(0, "%s", ip_addr);
+			}
+		}
```

Comment:
Can you use `inet_ntop(hent->h_addrtype, hent->addr_list[i], ip_addr, sizeof(ip_addr));` so no need for `if` `else`?